src/files/tsv.ts

```diff
@@ -2,44 +2,57 @@
  * TSV
  * Module for parsing TSV
  */
+import { TextLineStream } from '@std/streams'
 import { ColumnsMap } from '../types/columns.ts'
 import type { BIDSFile } from '../types/filetree.ts'
 import { filememoizeAsync } from '../utils/memoize.ts'
-import type { WithCache } from '../utils/memoize.ts'
+import { createUTF8Stream } from './streams.ts'
 
-const normalizeEOL = (str: string): string => str.replace(/\r\n/g, '\n').replace(/\r/g, '\n')
-// Typescript resolved `row && !/^\s*$/.test(row)` as `string | boolean`
-const isContentfulRow = (row: string): boolean => !!(row && !/^\s*$/.test(row))
+async function _loadTSV(file: BIDSFile, maxRows: number = -1): Promise<ColumnsMap> {
```

Comment:
This file in particular will be easiest to review just by reading the new file, as the contents are almost entirely new.